scripts/determine_tests_to_run.py

```diff
@@ -234,6 +250,25 @@ def determine_files_to_test(
     return sorted(res)
 
 
+def determine_tests_to_run(
+    test_candidates: List[str], ignore_paths: List[str], filter: Optional[str]
+) -> List[str]:
+    """
+    Applies a number of filters to a list of test candidates in order to determine
+    the final output to be fed to pytest.
+    """
```

Comment:
Little helper to do one last filtering; this takes into account `--ignore` and `--filter` opts that are used in the pipeline. Without these args, `pytest` has the tendency to test the entire suite.